### PR TITLE
Update transaction-lifecycle.md

### PR DESCRIPTION
**Pull request by laynelafrance**
initial draft for tx lifecycle

Input needed from: 
* protocol team on`Validation` `clustering` `verification` and `sealing` sections cc @franklywatson
* FVM team on `Execution` and `computation` cc @fotescodev @ramtinms

Closes: [slack discussion](https://dapperlabs.slack.com/archives/C01E563SA92/p1676309714522109)

## Description

______

For contributor use:

- [ ] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Updated relevant documentation 
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

**Comment by laynelafrance** [comment]
@janezpodhostnik are you able to jump in from an FVM perspective on the computation and execution sections above?

File: docs/content/concepts/transaction-lifecycle.md [hunk 1]
```diff
@@ -52,3 +54,13 @@ _Documentation coming soon..._
...
 
 _Documentation coming soon..._
+
+## Transaction Results
+From the [Access API](https://github.com/onflow/flow-go/blob/master/cmd/access/README.md)
+
+[GetTransactionResult](/nodes/access-api#gettransaction): an execution node is requested for events for the transaction and the transaction status is derived as follows:
+* If the collection containing the transaction and the block containing that collection is found locally, but the transaction has expired then its status is returned as `expired`.
+* If either the collection or the block is not found locally, but the transaction has not expired, then its status is returned as `pending`
+If the transaction has neither expired nor is it pending, but the execution node has not yet executed the transaction, then the status of the transaction is returned as `finalized`.
```
Comment (franklywatson) [inline docs/content/concepts/transaction-lifecycle.md:64]:
```suggestion
* If the transaction has neither expired nor is it pending, but the execution node has not yet executed the transaction, then the status of the transaction is returned as `finalized`.
```

File: docs/content/concepts/transaction-lifecycle.md [hunk 2]
```diff
@@ -52,3 +54,13 @@ _Documentation coming soon..._
...
+* If the execution node has executed the transaction, then if the height of the block containing the transaction is greater than the highest sealed block, then the status of the transaction is returned as `executed` else it is returned as `sealed`.
```
Comment (franklywatson) [inline docs/content/concepts/transaction-lifecycle.md:65]:
```suggestion
* If the execution node has executed the transaction, and if the height of the block containing the transaction is greater than the highest sealed block, then the status of the transaction is returned as `executed` else it is returned as `sealed`.
```

**Review by franklywatson — APPROVED** [review]
Minor typos

**Comment by janezpodhostnik** [comment]
@laynelafrance I can take a look later this week

**Pull request merged**